### PR TITLE
[configure.ac] implement SAI API version check

### PR DESCRIPTION
**Pull request by stepanblyschak**
Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>

The motivation for this change is described in the proposal https://github.com/Azure/SONiC/pull/935 and proposal in SAI https://github.com/opencomputeproject/SAI/pull/1404
NOTE: Requires to update SAI once https://github.com/opencomputeproject/SAI/pull/1404 is in.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -163,6 +163,29 @@ AC_SUBST(CXXFLAGS_COMMON)
...
+SAVED_FLAGS="$CXXFLAGS"
+CXXFLAGS="-lsai -I$srcdir/SAI/inc -I$srcdir/SAI/experimental -I$srcdir/SAI/meta"
+AC_MSG_CHECKING([SAI headers API version and library version check])
+AC_TRY_RUN([
+extern "C" {
+#include <sai.h>
+}
+int main() {
+    sai_api_version_t version;
+    if (SAI_STATUS_SUCCESS != sai_query_api_version(&version))
```
Comment (stepanblyschak) [inline configure.ac:177]: Also add a runtime check in syncd + log to syslog

**Comment by kcudnik** [comment]
SAI module needs to be updated first to support sai version type

**Comment by liat-grozovik** [comment]
@kcudnik kindly remonder to review this PR now that 1.10.x is intergated into SONiC master

**Review by kcudnik — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
https://github.com/Azure/sonic-sairedis/pull/1034 is merged

**Pull request merged**